From 7df52446a398ac3a7b5cc887dae6e1df532dc0bb Mon Sep 17 00:00:00 2001
From: Fang Hui <b31070@freescale.com>
Date: Tue, 10 Sep 2013 17:30:23 +0800
Subject: [PATCH 471/635] ENGR00278594: It takes 4s to take a photo of 5M
 Pixel

4.3 framework change quality type from int32 to uint8,
HAL need to sync.

Signed-off-by: Fang Hui <b31070@freescale.com>
---
 mx6/libcamera2/MetadaManager.cpp | 18 ++++++++++++++----
 1 file changed, 14 insertions(+), 4 deletions(-)

diff --git a/mx6/libcamera2/MetadaManager.cpp b/mx6/libcamera2/MetadaManager.cpp
index f09f373..3d01e82 100755
--- a/mx6/libcamera2/MetadaManager.cpp
+++ b/mx6/libcamera2/MetadaManager.cpp
@@ -233,7 +233,8 @@ status_t MetadaManager::createDefaultRequest(
     ADD_OR_SIZE(ANDROID_SCALER_CROP_REGION, cropRegion, 3);
 
     /** android.jpeg */
-    static const int32_t jpegQuality = 100;
+    //4.3 framework change quality type from i32 to u8
+    static const uint8_t jpegQuality = 100;
     ADD_OR_SIZE(ANDROID_JPEG_QUALITY, &jpegQuality, 1);
 
     static const int32_t thumbnailSize[2] = {
@@ -241,7 +242,8 @@ status_t MetadaManager::createDefaultRequest(
     };
     ADD_OR_SIZE(ANDROID_JPEG_THUMBNAIL_SIZE, thumbnailSize, 2);
 
-    static const int32_t thumbnailQuality = 100;
+    //4.3 framework change quality type from i32 to u8
+    static const uint8_t thumbnailQuality = 100;
     ADD_OR_SIZE(ANDROID_JPEG_THUMBNAIL_QUALITY, &thumbnailQuality, 1);
 
     static const double gpsCoordinates[3] = {
@@ -479,6 +481,7 @@ status_t MetadaManager::getJpegRotation(int32_t &jpegRotation)
 
 status_t MetadaManager::getJpegQuality(int32_t &quality)
 {
+    uint8_t u8Quality = 0;
     camera_metadata_entry_t streams;
     int res = find_camera_metadata_entry(mCurrentRequest,
                 ANDROID_JPEG_QUALITY, &streams);
@@ -487,12 +490,16 @@ status_t MetadaManager::getJpegQuality(int32_t &quality)
         return BAD_VALUE;
     }
 
-    quality = streams.data.i32[0];
+    //4.3 framework change quality type from i32 to u8
+    u8Quality = streams.data.u8[0];
+    quality = u8Quality;
+
     return NO_ERROR;
 }
 
 status_t MetadaManager::getJpegThumbQuality(int32_t &thumb)
 {
+    uint8_t u8Quality = 0;
     camera_metadata_entry_t streams;
     int res = find_camera_metadata_entry(mCurrentRequest,
                 ANDROID_JPEG_THUMBNAIL_QUALITY, &streams);
@@ -501,7 +508,10 @@ status_t MetadaManager::getJpegThumbQuality(int32_t &thumb)
         return BAD_VALUE;
     }
 
-    thumb = streams.data.i32[0];
+    //4.3 framework change quality type from i32 to u8
+    u8Quality = streams.data.u8[0];
+    thumb = u8Quality;
+
     return NO_ERROR;
 }
 
-- 
1.8.0

